src/deadline/job_attachments/api/manifest.py

```diff
@@ -247,3 +260,182 @@ def process_output(status: FileStatus, path: str, output_diff: ManifestDiff):
+        upload.upload_bytes_to_s3(
+            bytes=BytesIO(manifest.read().encode("utf-8")),
+            bucket=s3_bucket_name,
+            key=manifest_path,
+            progress_handler=logger.echo,
+            extra_args=s3_metadata,
+        )
+
+
+def _manifest_download(
+    download_dir: str,
+    farm_id: str,
+    queue_id: str,
+    job_id: str,
+    boto3_session: boto3.Session,
+    step_id: Optional[str] = None,
+    logger: ClickLogger = ClickLogger(False),
+) -> ManifestDownloadResponse:
+    """
+    BETA API - This API is still evolving but will be made public in the near future.
+    API to download the Job Attachment manifest for a Job, and optionally dependencies for Step.
+    download_dir: Download directory.
+    farm_id: The Deadline Farm to download from.
+    queue_id: The Deadline Queue to download from.
+    job_id: Job Id to download.
+    boto_session: Boto3 session.
+    step_id: Optional[str]: Optional, download manifest for a step
+    logger: Click Logger instance to print to CLI as test or JSON.
+    return ManifestDownloadResponse Downloaded Manifest data. Contains source S3 key and local download path.
+    """
+
+    # Deadline Client and get the Queue to download.
+    deadline = boto3_session.client("deadline", config=get_default_client_config())
+
+    queue: dict = deadline.get_queue(
+        farmId=farm_id,
+        queueId=queue_id,
+    )
+
+    # assume queue role - session permissions
+    queue_role_session: boto3.Session = _get_queue_user_boto3_session(
+        deadline=deadline,
+        base_session=boto3_session,
+        farm_id=farm_id,
+        queue_id=queue_id,
+        queue_display_name=queue["displayName"],
+    )
+
+    # Get S3 prefix
+    s3_prefix: Path = Path(queue["jobAttachmentSettings"]["rootPrefix"], S3_MANIFEST_FOLDER_NAME)
+
+    # Queue's Job Attachment settings.
+    queue_s3_settings = JobAttachmentS3Settings(**queue["jobAttachmentSettings"])
+
+    # Capture a list of success and failed to download files for JSON output.
+    successful_downloads: List[ManifestDownload] = []
+    failed_downloads: List[str] = []
+
+    # Utility function to build up manifests by root.
+    manifests_by_root: Dict[str, List[BaseAssetManifest]] = dict()
+
+    def add_manifest_by_root(
+        manifests_by_root: Dict[str, list], root: str, manifest: BaseAssetManifest
+    ):
+        if root not in manifests_by_root:
+            manifests_by_root[root] = []
+        manifests_by_root[root].append(manifest)
+
+    # Get input_manifest_paths from Deadline GetJob API
+    job: dict = deadline.get_job(farmId=farm_id, queueId=queue_id, jobId=job_id)
+    attachments: dict = job["attachments"]
+    input_manifest_paths: List[Tuple[str, str]] = [
+        (manifest["inputManifestPath"], manifest["rootPath"])
+        for manifest in attachments["manifests"]
+    ]
+
+    # Download each input_manifest_path
+    for input_manifest in input_manifest_paths:
```

Comment:
nit: for readability, it would be nice if we split up the tuple into individual variables here
```suggestion
    for input_manifest_path, root_path in input_manifest_paths:
```

Reply:
Great idea! (one of these pythonic things I'll never fully get used to)